e2e/stack_monitoring_test.go

```diff
@@ -0,0 +1,614 @@
+		"ecs",
+		"metricset",
+	}
+	allowedInsertionsInMetricbeatDocs = append(allowedInsertionsInMetricbeatDocs, sm.allowedInsertionsExtra...)
+
+	allowedDeletionsFromMetricbeatDocs := []string{
+		"source_node",
+	}
+	allowedDeletionsFromMetricbeatDocs = append(allowedDeletionsFromMetricbeatDocs, sm.allowedDeletionsExtra...)
+
+	hitsPath := "hits.hits"
+	legacyHits := legacyContainer.Path(hitsPath)
+	metricbeatHits := metricbeatContainer.Path(hitsPath)
+
+	legacyTypes, legacySources := checkSourceTypes(legacyHits)
+	metricbeatTypes, metricbeatSources := checkSourceTypes(metricbeatHits)
+
+	if len(legacyTypes) > len(metricbeatTypes) {
+		// returns an array as the result of removing all elements in 'b' from the 'a' array
+		// only used here
+		arrayDiff := func(a []string, b []string) []string {
+			target := map[string]bool{}
+			for _, x := range b {
+				target[x] = true
+			}
+
+			result := []string{}
+			for _, x := range a {
+				if _, ok := target[x]; !ok {
+					result = append(result, x)
+				}
+			}
+
+			return result
+		}
+
+		diff := arrayDiff(legacyTypes, metricbeatTypes)
+
+		return fmt.Errorf("Found more legacy-indexed document types than metricbeat-indexed document types. Document types indexed by internal collection but not by Metricbeat collection: %v", diff)
+	}
+
+	errors := []error{}
+	for docType, sourceValue := range legacySources {
+		legacyDoc := gabs.Wrap(sourceValue)
+		metricbeatDoc := gabs.Wrap(metricbeatSources[docType])
+
+		err := sm.handleSpecialCases(docType, legacyDoc, metricbeatDoc)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		unexpectedInsertions := []string{}
+		unexpectedDeletions := []string{}
+
+		differences := jsonDiff(legacyDoc, metricbeatDoc)
+
+		lines := strings.Split(differences, "\n")
+		for _, line := range lines {
+			trimmedLine := strings.TrimSpace(line)
+
+			if !strings.HasPrefix(trimmedLine, `"ADDED":`) && !strings.HasPrefix(trimmedLine, `"REMOVED":`) {
+				continue
+			}
+
+			if strings.HasPrefix(trimmedLine, `"ADDED": `) {
+				if !strings.HasSuffix(trimmedLine, ",") {
+					continue
+				} else if checkAllowedField(trimmedLine, allowedInsertionsInMetricbeatDocs) {
+					continue
+				} else if strings.HasSuffix(trimmedLine, "{} (object)},") {
+					continue
+				}
+
+				unexpectedInsertions = append(unexpectedInsertions, trimmedLine)
+				continue
+			}
+
+			if strings.HasPrefix(trimmedLine, `"REMOVED": `) {
+				if !strings.HasSuffix(trimmedLine, ",") {
+					continue
+				} else if checkAllowedField(trimmedLine, allowedDeletionsFromMetricbeatDocs) {
+					continue
+				} else if strings.HasSuffix(trimmedLine, "{} (object)},") {
+					continue
+				}
+
+				unexpectedDeletions = append(unexpectedDeletions, trimmedLine)
+				continue
+			}
+		}
+
+		if len(unexpectedInsertions) == 0 && len(unexpectedDeletions) == 0 {
+			log.WithFields(log.Fields{
+				"docType": docType,
+			}).Debugf("Expected parity between Metricbeat-indexed doc and legacy-indexed doc found.")
```

Comment:
It's nice to know when the tests pass. How about we move this to INFO level and also mention the product name and `docType`?